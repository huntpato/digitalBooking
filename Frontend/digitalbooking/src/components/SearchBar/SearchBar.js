import React, { Fragment, useMemo, useState } from 'react'
import { Button2 } from '../Button2/Button2'
import DatePicker2 from '../DatePicker2/DatePicker2'
import { DateContext } from './DateContext';
import { LocationContext } from './LocationContext';
import { Placepicker } from '../PlacePicker/Placepicker';
import "react-datepicker/dist/react-datepicker.css";
import styles from './SearchBar.module.css'

export const SearchBar = props => {

    const{search} = styles;
    const [date, setDate] = useState(false);
    const [place, setPlace] = useState(false);
    const value = useMemo(() => ({ date, setDate }), [date, setDate ]);
    const value2 = useMemo(() => ({ place, setPlace }), [place, setPlace ]);

    const handleContext= ()=>{
        localStorage.setItem('date', date);
        localStorage.setItem('place', place);
    }


    return (
        <Fragment>
            <div className={search}>
                <p>Busca ofertas en hoteles, casas y mucho m√°s</p>
                <section>
                    <LocationContext.Provider value={value2}>
                    <Placepicker />
                    </LocationContext.Provider>
                    <DateContext.Provider value={value}>
                    <DatePicker2 dateFormat="dd/MM/yyyy" />
                    </DateContext.Provider>
                    <Button2 handleClick= {handleContext} text="Buscar" link='/product'/>
                </section>
            </div>
        </Fragment>
    )
}

