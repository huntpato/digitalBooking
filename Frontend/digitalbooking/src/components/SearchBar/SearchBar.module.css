.search {
  align-items: center;
  background-color: var(--logo-color);
  color: var(--background-white);
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  font-size: 18px;
  font-weight: 700;
  height: 111px;
  padding-bottom: 10px;
  padding-top: 20px;
  width: 100%;
}

.search section{
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content:center;
    width: 60%;

}

.search p{
    padding-bottom: 10px;
}

@media (max-width: 768px){

  .search section{

    width: 70%;

}
}

@media (max-width: 576px) {

  .search{
    height: 200px;
}

  .search section{
    align-items: center;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    width: 90vw;
  }

  .search p{
    font-size: 16px;
  }

}

@media (max-width: 480px){

    .search{
        height: 200px;
    }

    input:focus {
      font-size: 16px;
    }
    
    .search section{
        align-items: center;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        justify-content: center;
        width: 90vw;
    }

    .search p{
      font-size: 16px;
    }

   
}
