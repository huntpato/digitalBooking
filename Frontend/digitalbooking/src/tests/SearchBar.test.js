import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from '../components/SearchBar/SearchBar'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <SearchBar />', ()=> {

    const wrapper= shallow(<SearchBar />);

    test('Comprobar que <SearchBar /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })


    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('search')).toBe(true);
    })

})
