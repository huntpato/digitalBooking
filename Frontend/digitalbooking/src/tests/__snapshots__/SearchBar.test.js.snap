// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <SearchBar /> Comprobar que <SearchBar /> renderiza correctamente. 1`] = `
<Fragment>
  <div
    className="search"
  >
    <p>
      Busca ofertas en hoteles, casas y mucho m√°s
    </p>
    <section>
      <ContextProvider
        value={
          Object {
            "place": false,
            "setPlace": [Function],
          }
        }
      >
        <Placepicker />
      </ContextProvider>
      <ContextProvider
        value={
          Object {
            "date": false,
            "setDate": [Function],
          }
        }
      >
        <DatePicker2
          dateFormat="dd/MM/yyyy"
        />
      </ContextProvider>
      <Button2
        handleClick={[Function]}
        link="/product"
        text="Buscar"
      />
    </section>
  </div>
</Fragment>
`;
